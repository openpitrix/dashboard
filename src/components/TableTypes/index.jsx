import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import _ from 'lodash';
import { translate } from 'react-i18next';

import styles from './index.scss';

@translate()
export default class TableTypes extends Component {
  static propTypes = {
    activeType: PropTypes.string,
    changeType: PropTypes.func,
    className: PropTypes.string,
    types: PropTypes.array
  };

  static defaultProps = {
    changeType: _.noop,
    type: [],
    activeType: ''
  };

  constructor(props) {
    super(props);

    const activeType = _.get(props.types, '[0].value', '');
    this.state = {
      activeType: props.activeType || activeType
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    return nextState.activeType !== this.state.activeType;
  }

  componentDidUpdate() {
    this.props.changeType(this.state.activeType);
  }

  handleChange = type => {
    this.setState({ activeType: type });
  };

  render() {
    const { className, types, t } = this.props;
    const { activeType } = this.state;

    return (
      <div className={classnames(styles.tableTypes, className)}>
        {types.map(type => (
          <label
            key={type.value}
            onClick={() => this.handleChange(type.value)}
            className={classnames({
              [styles.active]: activeType === type.value
            })}
          >
            {t(type.name)}
          </label>
        ))}
      </div>
    );
  }
}
