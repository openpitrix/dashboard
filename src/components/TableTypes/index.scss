@import "~scss/vars";

.tableTypes {
  margin-bottom: 16px;

  > label {
    display: inline-block;
    padding: 9px 12px;
    border-radius: 2px;
    background-color: $N0;
    border: 1px solid $N10;
    font-size: 14px;
    line-height: 14px;
    color: $N300;
    cursor: pointer;

    &.active {
      color: $P75;
      border-color: $P75;
    }
  }
}
