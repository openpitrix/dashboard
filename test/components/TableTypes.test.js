import TableTypes from 'components/TableTypes';

const types = [
  { name: 'Unreviewed', value: 'unreviewed', status: ['submitted'] },
  { name: 'Reviewed', value: 'reviewed', status: ['rejected', 'passed'] }
];

describe('Component TableTypes', () => {
  it('basic render', () => {
    const wrapper = render(<TableTypes types={types} activeType="reviewed" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
