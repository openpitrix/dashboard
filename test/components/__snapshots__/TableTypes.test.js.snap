// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TableTypes basic render 1`] = `
<div
  class="tableTypes"
>
  <label
    class=""
  >
    Unreviewed
  </label>
  <label
    class="active"
  >
    Reviewed
  </label>
</div>
`;
